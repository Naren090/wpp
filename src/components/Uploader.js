import React, { Component, PropTypes } from 'react';
import Dropzone from 'react-dropzone';
import { List, ListItem } from 'material-ui/List';
import RaisedButton from 'material-ui/RaisedButton';
import ActionDelete from 'material-ui/svg-icons/action/delete';
import Divider from 'material-ui/Divider';
import Checkbox from 'material-ui/Checkbox';

const style = {
	width: '100%',
	height: 103,
	borderWidth: 2,
	borderColor: '#666',
	borderStyle: 'dashed',
	borderRadius: 5,
	margin: '8px 0'
};

export class Uploader extends Component {
	constructor(props) {
		super(props);
		this.state = {
			data: {
				files: [],
				ZipCheck: false,
				PDFCheck: false,
			},
			isZipFileAvailable: false,
			isPDFFileAvailable: false,
			multiple: props.multiple,

		};
		this.onDrop = this.onDrop.bind(this);
		this.onOpenClick = this.onOpenClick.bind(this);
		this.onZipCheck = this.onZipCheck.bind(this);
		this.onPDFCheck = this.onPDFCheck.bind(this);
		this.deleteFile = this.deleteFile.bind(this);
	}
	onDrop(acceptedFiles) {
		const files = [...this.state.data.files, ...acceptedFiles];
		const data = { ...this.state.data, files };
		this.setState({
			isZipFileAvailable: data.files.find(x => x.type === "application/x-zip-compressed" || x.type === "application/zip") !== undefined,
			isPDFFileAvailable: data.files.find(x => x.type === "application/pdf") !== undefined,
			data
		});
		this.props.onDataChange(data);
	}
	onOpenClick() {
		this.dropzone.open();
	}
	onZipCheck(e, isInputChecked) {
		const data = { ...this.state.data, ZipCheck: !isInputChecked };
		this.setState({ data });
		this.props.onDataChange(data);
	}
	onPDFCheck(e, isInputChecked) {
		const data = { ...this.state.data, PDFCheck: !isInputChecked };
		this.setState({ data });
		this.props.onDataChange(data);
	}

	deleteFile(file) {
		const files = this.state.data.files.filter(obj => obj !== file);
		const data = { ...this.state.data, files };
		this.setState({
			isZipFileAvailable: data.files.find(x => x.type === "application/x-zip-compressed" || x.type === "application/zip") !== undefined,
			isPDFFileAvailable: data.files.find(x => x.type === "application/pdf") !== undefined,
			data
		});
		this.props.onDataChange(data);
	}

	render() {

		return (
			<div>
				<Dropzone ref={(node) => { this.dropzone = node; }} onDrop={this.onDrop} multiple={this.state.multiple} style={style}>
					<div style={{ display: 'flex', flexDirection: 'column', justifyContent: 'center', alignItems: 'center', padding: 8 }} >
						<p>Drag and drop assets here</p>
						<RaisedButton primary={true} label="Browse File" />
					</div>
				</Dropzone>
				{
					this.state.isZipFileAvailable ? <Checkbox label="Extract zip content" checked={this.state.data.ZipCheck} onCheck={this.onZipCheck} /> : null
				}
				{
					this.state.isPDFFileAvailable ? <Checkbox label="Display PDF in viewer" checked={this.state.data.PDFCheck} onCheck={this.onPDFCheck} /> : null
				}
				<div style={this.state.data.files.length > 0 ? { height: 150, overflowX: 'auto' } : {}}>
					{this.state.data.files.length > 0 ? <List style={{ margin: 8 }}>{this.state.data.files.map((file, i) => { return <div key={i}><ListItem primaryText={file.name} rightIcon={<ActionDelete onClick={() => this.deleteFile(file)} />} /><Divider /></div>; })}</List> : null}
				</div>
			</div>
		);
	}
}

Uploader.propTypes = {
	onDataChange: PropTypes.func.isRequired,
	multiple: PropTypes.bool,
};

export default Uploader;