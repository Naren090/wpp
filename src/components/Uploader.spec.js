import React from 'react';
import { shallow } from 'enzyme';
import { Uploader } from './Uploader';

function setupShallow() {
	const props = {
		multiple: false,
		onDataChange: jest.fn()
	};
	const enzymeWrapper = shallow(<Uploader {...props} />);
	return {
		props,
		enzymeWrapper
	};
}


describe('Uploader', () => {

	it('should render self and subcomponents', () => {
		const { enzymeWrapper, props } = setupShallow();

		const dropzon = enzymeWrapper.find('Dropzone').props();
		expect(dropzon.multiple).toBe(props.multiple);

		expect(enzymeWrapper.find('Dropzone').childAt(0).is('div')).toBe(true);
		expect(enzymeWrapper.find('Dropzone').childAt(0).is('div')).toBe(true);
		expect(enzymeWrapper.find('Dropzone').childAt(0).childAt(0).is('p')).toBe(true);
		expect(enzymeWrapper.find('Dropzone').childAt(0).childAt(1).is('RaisedButton')).toBe(true);

	});

	it('should render Extract zip content', () => {
		const { enzymeWrapper } = setupShallow();
		enzymeWrapper.setState({ isZipFileAvailable: true });
		const zipChekbox = enzymeWrapper.find('Checkbox').props();
		expect(zipChekbox.label).toBe('Extract zip content');
	});

	it('should render Display PDF in viewer', () => {
		const { enzymeWrapper } = setupShallow();
		enzymeWrapper.setState({ isPDFFileAvailable: true });
		const pdfChekbox = enzymeWrapper.find('Checkbox').props();
		expect(pdfChekbox.label).toBe('Display PDF in viewer');
	});

	it('should render both checkbox', () => {
		const { enzymeWrapper } = setupShallow();
		enzymeWrapper.setState({ isZipFileAvailable: true, isPDFFileAvailable: true });
		const zipChekbox = enzymeWrapper.find('Checkbox').at(0).props();
		const pdfChekbox = enzymeWrapper.find('Checkbox').at(1).props();
		expect(zipChekbox.label).toBe('Extract zip content');
		expect(pdfChekbox.label).toBe('Display PDF in viewer');
	});
});

