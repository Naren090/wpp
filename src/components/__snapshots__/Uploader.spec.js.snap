exports[`Uploader should render children 1`] = `
"<div><div class="" style="width: 200px; height: 200px; border-width: 2px; border-color: #666; border-style: dashed; border-radius: 5px;"><div style="display: flex; padding: 8px;"><p>Drag and drop assets here</p><div style="color: rgba(0, 0, 0, 0.87); background-color: rgb(255, 255, 255); box-sizing: border-box; font-family: Roboto, sans-serif; box-shadow: 0 1px 6px rgba(0, 0, 0, 0.12),
         0 1px 4px rgba(0, 0, 0, 0.12); border-radius: 2px; display: inline-block; min-width: 88px;"><button tabindex="0" type="button" style="border: 10px; box-sizing: border-box; display: inline-block; font-family: Roboto, sans-serif; cursor: pointer; text-decoration: none; margin: 0px; padding: 0px; outline: none; font-size: inherit; font-weight: inherit; position: relative; z-index: 1; height: 36px; line-height: 36px; width: 100%; border-radius: 2px; background-color: rgb(0, 188, 212); text-align: center;"><div><div style="height: 36px; border-radius: 2px; top: 0px;"><span style="position: relative; opacity: 1; font-size: 14px; letter-spacing: 0; text-transform: uppercase; font-weight: 500; margin: 0px; padding-left: 16px; padding-right: 16px; color: rgb(255, 255, 255);">Browse File</span></div></div></button></div></div><input type="file" multiple="" style="display: none;"></div></div>"
`;
